### PR TITLE
feat(index): expose createURL

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

This is useful for recreating what an index widget does when it renders, see ssr in Vue, as well as poc/telemetry2

I think in the future we might want a function to retrieve the parameters given to render / init.

**Result**

index exposes the scoped `createURL` function to be used in render / init / getWidgetRenderState etc.

### CODE REVIEW

**Comment by Haroenv** [comment]
just realised the tests didn't get committed, let's add something now

File: src/widgets/index/__tests__/index-test.ts [hunk 1]
```diff
@@ -651,6 +652,103 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/index-widge
...
+      const instance = index({ indexName: 'indexName' });
+      const searchBox = createSearchBox();
+      const pagination = createPagination();
+
+      instance.addWidgets([searchBox, pagination]);
+
+      instance.init(
+        createInitOptions({
+          instantSearchInstance: createInstantSearch({
+            // @ts-ignore
```
Comment (shortcuts) [inline src/widgets/index/__tests__/index-test.ts:678]: Is it still needed as it's called at L679?

**Review by shortcuts — APPROVED** [review]
LGTM! 🥇

File: src/widgets/index/__tests__/index-test.ts (hunk 1 above)
Comment (Haroenv) [reply src/widgets/index/__tests__/index-test.ts:678]: yes, this function is meant to return a string, but an object is easier to assert on

**Pull request merged**